6.0.1 - macOS fails to compile `rpmio/lposix.cc`

**Describe the bug**
macOS fails to compile `rpmio/lposix.cc` from code at:
https://github.com/rpm-software-management/rpm/blob/4ea8622a8ffc2a5c24aaca754fd268f5d0b7032b/rpmio/lposix.cc#L425-L427

Error may have started since switch from C to C++.

Moving above code out of function seems to fix issue. Or using the `misc/system.h` environ but I haven't checked if that has any unintended side effects.

**To Reproduce**
Steps to reproduce the behavior:
1. Will need a macOS machine as specific to `__APPLE__` code
2. `cmake -S . -B _build -DENABLE_NLS=ON -DENABLE_PLUGINS=OFF -DWITH_AUDIT=OFF -DWITH_SELINUX=OFF -DENABLE_TESTSUITE=OFF -DWITH_ACL=OFF -DWITH_CAP=OFF -DWITH_LIBELF=OFF -DWITH_LIBDW=OFF && cmake --build _build`
3. Error encountered

**Expected behavior**
Successful build. 

**Output**
```
In file included from /private/tmp/rpm-20260308-63363-slerxg/rpm-6.0.1/rpmio/lposix.cc:426:
/Library/Developer/CommandLineTools/SDKs/MacOSX26.sdk/usr/include/crt_externs.h:39:1: error: expected unqualified-id
   39 | __BEGIN_DECLS
      | ^
/Library/Developer/CommandLineTools/SDKs/MacOSX26.sdk/usr/include/sys/cdefs.h:71:32: note: expanded from macro '__BEGIN_DECLS'
   71 | #define __BEGIN_DECLS   extern "C" {
      |                                ^
/private/tmp/rpm-20260308-63363-slerxg/rpm-6.0.1/rpmio/lposix.cc:432:8: error: use of undeclared identifier '_NSGetEnviron'
  432 |                 if (*environ==NULL) lua_pushnil(L); else lua_newtable(L);
      |                      ^
/private/tmp/rpm-20260308-63363-slerxg/rpm-6.0.1/rpmio/lposix.cc:427:21: note: expanded from macro 'environ'
  427 |                 #define environ (*_NSGetEnviron())
      |                                   ^
/private/tmp/rpm-20260308-63363-slerxg/rpm-6.0.1/rpmio/lposix.cc:433:10: error: use of undeclared identifier '_NSGetEnviron'
  433 |                 for (e=environ; *e!=NULL; e++)
      |                        ^
/private/tmp/rpm-20260308-63363-slerxg/rpm-6.0.1/rpmio/lposix.cc:427:21: note: expanded from macro 'environ'
  427 |                 #define environ (*_NSGetEnviron())
      |                                   ^
3 errors generated.
```

**Environment**
 - OS / Distribution: macOS 26 Tahoe
 - Version: rpm 6.0.1

**Additional context**
Seen while trying to update Homebrew's package of `rpm` to latest version.

--- a/rpmio/lposix.cc
+++ b/rpmio/lposix.cc
@@ -5,9 +5,7 @@
 * 05 Nov 2003 22:09:10
 */
 
-#ifdef HAVE_CONFIG_H
-#include <config.h>
-#endif
+#include "system.h"
 
 #include <vector>
 
@@ -42,6 +40,8 @@
 #endif
 
 #include "modemuncher.cc"
+
+#include "debug.h"
 
 extern int _rpmlua_have_forked;
 
@@ -422,12 +422,6 @@ static int Pgetenv(lua_State *L)		/** getenv([name]) */
 {
 	if (lua_isnone(L, 1))
 	{
-	#ifdef __APPLE__
-		#include <crt_externs.h>
-		#define environ (*_NSGetEnviron())
-	#else
-		extern char **environ;
-	#endif /* __APPLE__ */
 		char **e;
 		if (*environ==NULL) lua_pushnil(L); else lua_newtable(L);
 		for (e=environ; *e!=NULL; e++)

